### PR TITLE
PERF-645 Separate ad part model based on type, validation of type

### PR DESCRIPTION
**Pull request by olga-andreeva-gyg**

### CODE REVIEW

File: tests/ad_templates/factories.py [hunk 1]
```diff
@@ -15,15 +16,50 @@ class Meta:
 
 class AdPartFactory(BaseSQLAlchemyModelFactory):
     class Meta:
+        abstract = True
         model = AdPart
 
     template_text = factory.Sequence(lambda n: f'text template {n}')
-    template_type = random.choice(AdComponent.ALL_COMPONENTS)
```
Comment (andrebco) [inline tests/ad_templates/factories.py]: Why not keeping this here for the cases that we don't care about `template_type`.

File: tests/ad_templates/models/test_get_one_or_create.py [hunk 2]
```diff
@@ -8,23 +9,23 @@ class TestGetOneOrCreate(BaseTestCase):
...
-            headline1_id=AdPartFactory(template_text="headline_1").id,
-            headline2_id=AdPartFactory(template_text="headline_2").id,
-            description_id=AdPartFactory(template_text="description").id,
-            path1_id=AdPartFactory(template_text="path_1").id,
-            path2_id=AdPartFactory(template_text="path_2").id,
+            headline1_id=Headline1Factory(template_text="headline_1").id,
+            headline2_id=Headline2Factory(template_text="headline_2").id,
+            description_id=DescriptionFactory(template_text="description").id,
+            path1_id=Path1Factory(template_text="path_1").id,
+            path2_id=Path2Factory(template_text="path_2").id,
```
Comment (andrebco) [inline tests/ad_templates/models/test_get_one_or_create.py]: If we are having these `Factories`, why not having the `template_text` to be automatically generated?

File: tests/ad_templates/models/test_get_one_or_create.py [hunk 3]
```diff
@@ -8,23 +9,23 @@ class TestGetOneOrCreate(BaseTestCase):
...
-        headline1 = AdPartFactory(template_text="headline_1")
-        headline2 = AdPartFactory(template_text="headline_2")
-        description = AdPartFactory(template_text="description")
-        path1 = AdPartFactory(template_text="path_1")
-        path2 = AdPartFactory(template_text="path_2")
+        headline1 = Headline1Factory(template_text="headline_1")
+        headline2 = Headline2Factory(template_text="headline_2")
+        description = DescriptionFactory(template_text="description")
+        path1 = Path1Factory(template_text="path_1")
+        path2 = Path2Factory(template_text="path_2")
```
Comment (andrebco) [inline tests/ad_templates/models/test_get_one_or_create.py]: Same here. We could clean the `template_text` values.

File: tests/ad_templates/factories.py (hunk 1 above)
Comment (olga-andreeva-gyg) [reply tests/ad_templates/factories.py]: We should always care about `template_type`. If there is no corresponding entry in one of `headline1`, `headline2`, ... tables, `AdPart` does not have any type and is therefore invalid.

File: tests/ad_templates/models/test_get_one_or_create.py (hunk 2 above)
Comment (olga-andreeva-gyg) [reply tests/ad_templates/models/test_get_one_or_create.py]: Done.

File: tests/ad_templates/factories.py (hunk 1 above)
Comment (andrebco) [reply tests/ad_templates/factories.py]: Ok. Gotha.

**Comment by olga-andreeva-gyg** [comment]
@olel-gyg do you think it is really needed? with the amount of data we have now i think we can just clean the database.

**Pull request merged**